sound/soc/soc-topology.c

```diff
@@ -373,8 +373,8 @@ static void remove_mixer(struct snd_soc_component *comp,
 		p = dobj->control.kcontrol->tlv.p;
 	snd_ctl_remove(card, dobj->control.kcontrol);
 	list_del(&dobj->list);
-	kfree(sm);
-	kfree(p);
+	devm_kfree(comp->dev, sm);
+	devm_kfree(comp->dev, p);
```

Comment:
removing superfluous code is good, so in general this series looks quite good, but let's merge this and the following patch. There's really no need to add `devm_kfee()` calls where it can be verified, that the resource will be freed. And doing it in two steps makes reviewing more difficult, which leads to the kind of mistakes, that I comment on later. Such a split might make this patch look simpler because it is kind of a mechanic 1-to-1 API replacement, but for that the following patch becomes much more difficult to verify.